### PR TITLE
Refactor scan all page UI

### LINKED ISSUE #9744: Show the scan all progress metrics in the UI side
Screen Shot 2019-11-05 at 17 06 46

### PR DESCRIPTION
**Pull request by AllForNothing**
Fix  #9744 and #9736
1.New UI
image
2. UI design: https://www.figma.com/proto/0bSkyzvchcOCrwSWROIivtIb/Harbor-v1.9?node-id=1251%3A1888&viewport=639%2C1697%2C0.5&scaling=min-zoom
3. Temporary available env. : https://10.160.194.191

### CODE REVIEW

File: src/portal/lib/src/config/config.ts [hunk 1]
```diff
@@ -159,3 +159,14 @@ export class Configuration {
...
+
+export class ScanningMetrics {
+    total?: number;
+    completed?: number;
+    metrics: {
+        [key: string]: number;
+    };
+    requester?: string;
+    isScheduled?: boolean;
+    ongoing: boolean;
```
Comment (zhoumeina) [inline src/portal/lib/src/config/config.ts]: "g" should be capital letter?

File: src/portal/lib/src/config/vulnerability/vulnerability-config.component.scss [hunk 2]
```diff
@@ -48,4 +48,53 @@
...
+        background-color: #e62700;
+    }
+    .finished {
+        background-color: #62a420;
+    }
+    .in-progress {
+        background-color: #0079b8;
+    }
+}
+.state {
```
Comment (zhoumeina) [inline src/portal/lib/src/config/vulnerability/vulnerability-config.component.scss]: .state {
    flex: 2;
    padding-left: 3px;
}
.value {
    flex: 3;
}

File: src/portal/lib/src/config/vulnerability/vulnerability-config.component.ts [hunk 3]
```diff
@@ -1,64 +1,93 @@
...
         private errorHandler: ErrorHandler,
         private translate: TranslateService,
-        private systemInfoService: SystemInfoService,
-        private configService: ConfigurationService
     ) { }
 
+    get scanningMetrics(): ScanningMetrics {
+        return this._scanningMetrics;
+    }
+    set scanningMetrics(metrics: ScanningMetrics) {
```
Comment (zhoumeina) [inline src/portal/lib/src/config/vulnerability/vulnerability-config.component.ts]: This scanningMetrics method should be tested. Lots of branches here.

File: src/portal/lib/src/config/config.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/lib/src/config/config.ts]: stay the same with back-end

File: src/portal/src/i18n/lang/zh-cn-lang.json [hunk 4]
```diff
@@ -875,16 +875,28 @@
...
             "DB_NOT_READY": "缺陷数据库可能没有完全准备好!",
-            "NEXT_SCAN": "下次可用时间"
+            "NEXT_SCAN": "下次可用时间",
+            "STATUS": {
+                "PENDING": "等待",
+                "RUNNING": "扫描中",
+                "STOPPED": "中止",
+                "ERROR": "出错",
+                "SUCCESS": "成功",
+                "SCHEDULED": "已入计划"
```
Comment (jwangyangls) [inline src/portal/src/i18n/lang/zh-cn-lang.json]: not sure '已入计划'

File: src/portal/lib/src/config/vulnerability/vulnerability-config.component.scss (hunk 2 above)
Comment (AllForNothing) [reply src/portal/lib/src/config/vulnerability/vulnerability-config.component.scss]: modified

**Comment by zhoumeina** [comment]
Some issues I found in the UI :
1 When scan is triggered, the running progress bar will disappear for a while and then appear. Looks strange. 
image
2 Sometimes when scan is triggered it report a in progress fail error. 

image

File: src/portal/lib/src/config/vulnerability/vulnerability-config.component.ts (hunk 3 above)
Comment (AllForNothing) [reply src/portal/lib/src/config/vulnerability/vulnerability-config.component.ts]: More cases have been added.

File: src/portal/src/i18n/lang/zh-cn-lang.json (hunk 4 above)
Comment (AllForNothing) [reply src/portal/src/i18n/lang/zh-cn-lang.json]: the status "scheduled" means the image is scheduled. '已入计划' should be ok

**Comment by AllForNothing** [comment]
> Some issues I found in the UI :
> 1 When scan is triggered, the running progress bar will disappear for a while and then appear. Looks strange.
> image
> 2 Sometimes when scan is triggered it report a in progress fail error.
> 
> image

Confirmed with back-end,these meet the design

**Pull request merged**